### PR TITLE
[DRAFT] Add migration with instance of delegation defined

### PR DESCRIPTION
**Pull request by paulperegud**
Hashes were computed using production values of SALTs

### CODE REVIEW

File: backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+
+def upgrade():
+    query = f"INSERT INTO score_delegation (hashed_addr) VALUES ('{hash1}');"
+    print(query, flush=True)
+    op.execute(query)
+    query = f"INSERT INTO score_delegation (hashed_addr) VALUES ('{hash2}');"
+    print(query, flush=True)
+    op.execute(query)
+    query = f"INSERT INTO score_delegation (hashed_addr) VALUES ('{hash3}');"
```
Comment (kgarbacinski) [inline backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py:33]: shouldn't we also save the antysibil score in the database after such delegations?

File: backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+"""
+from alembic import op
+# from app.modules.common.delegation import get_hashed_addresses
+
+revision = "999999999999"
+down_revision = "19f6b4f5ce2e"
+branch_labels = None
+depends_on = None
+
+# primary = "0x2222222222222222222222222222222222222222" # redacted
```
Comment (kgarbacinski) [inline backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py:16]: I think you can remove these commended out things

File: package.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
     "apitest:up": "docker compose -p apitest -f ./localenv/docker-compose.yaml -f ./localenv/apitest.yaml up anvil ipfs graph-postgres graph-node multideployer",
     "apitest:down": "docker compose -p apitest -f ./localenv/docker-compose.yaml -f ./localenv/apitest.yaml down",
     "apitest:run": "docker compose -p apitest -f ./localenv/docker-compose.yaml -f ./localenv/apitest.yaml run backend-apitest",
-    "preapitest:up": "docker rm -v -f $(docker ps -qa --filter 'name=apitest')",
```
Comment (kgarbacinski) [inline package.json:25]: why we need it?

File: backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py (hunk 1 above)
Comment (paulperegud) [reply backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py:29]: fixed

File: backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py (hunk 2 above)
Comment (paulperegud) [reply backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py]: +1

File: package.json (hunk 3 above)
Comment (paulperegud) [reply package.json]: Removed, will add in a separate PR.

**Comment by paulperegud** [comment]
Oh, it is still incomplete. anitysybil_score also needs to be updated.

**Comment by aziolek** [comment]
@paulperegud: :point_up:. E2Es fail on application readiness, probably because of migrations. Please take a look.

**Review by adam-gf — DISMISSED** [review]
🚀

File: backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+
+Revision ID: 999999999999
+Revises: 5d3bf271b78d
+Create Date: 2024-07-17 16:48:12.133033
+
+"""
+import json
+from alembic import op
+
+revision = "999999999999"
```
Comment (kgarbacinski) [inline backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py:11]: why this strange revision number?

File: backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+# (hash1, hash2, hash3) = get_hashed_addresses(primary, secondary)
+# values of hashes below were obtained using code above; with appropriately set DELEGATION_SALT and DELEGATION_SALT_PRIMARY envs
+hash1 = "a4ed43527ceb61080a61e8661aa63fa7a266a313c08473db31fa1acafa32bd16"
+hash2 = "772d00c482a244e447c8d460661b552fb9b06d58e7903f3bbfa4e0e14dd92b99"
+hash3 = "fb610f25ff00d710de6578a2b684fdd2dcb353ed341e9f22a100af754ef36ae6"
+
+
+def upgrade():
+    query = f"INSERT INTO score_delegation (hashed_addr) VALUES ('{hash1}');"
+    print(query, flush=True)
```
Comment (kgarbacinski) [inline backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py:25]: unnecessary print statements

File: backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+    op.execute(query)
+    query = f"INSERT INTO score_delegation (hashed_addr) VALUES ('{hash2}');"
+    print(query, flush=True)
+    op.execute(query)
+    query = f"INSERT INTO score_delegation (hashed_addr) VALUES ('{hash3}');"
+    print(query, flush=True)
+    op.execute(query)
+
+    # since sqlite doesn't support conditional inserts that well...
+    if op.get_bind().engine.name == "sqlite":
```
[the remark this answers is not in the document]
Comment (paulperegud) [reply backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py:35]: Our Dev env uses it (DevConfig). You can run it against sqlite using `flask db upgrade` locally. And this is not the first check like this in our migrations.

File: backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py (hunk 4 above)
Comment (paulperegud) [reply backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py:11]: I'm just really bad at generating random numbers ;-)

File: backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py (hunk 5 above)
Comment (paulperegud) [reply backend/migrations/versions/999999999999_manually_add_delegation_for_one_of_users.py:25]: +1

**Review by kgarbacinski — APPROVED** [review]
lgtm

**Comment by paulperegud** [comment]
It is not clear if this change is actually needed. Status TBD

**Comment by aziolek** [comment]
I'm closing this one. The user actually has score over 21, so delegation is neither required, nor supported, nor allowed.

**Pull request closed without merging**